src/http_server.cc

```diff
@@ -4105,8 +4163,15 @@ HTTPAPIServer::GenerateRequestClass::EndResponseCallback(
   auto infer_request =
       reinterpret_cast<HTTPAPIServer::GenerateRequestClass*>(arg);
 
-  infer_request->SendChunkResponse(true /* end */);
-  evhtp_send_reply_chunk_end(infer_request->EvHtpRequest());
+  if (InferRequestClass::active_requests_.count(infer_request->req_) == 0) {
```

Comment:
I think this is introducing a check in the hot path of inferencing. Is there any way to avoid requiring this check?

Reply:
check is on response sending - and I don't think there is anyway to avoid it - as the basic issue is whether or not the evhtp request has been freed. So we have to check before using it.

Note: this could have happened even in non generative case I believe because if a request took a long time to process and the client connection was freed we would have also run into this.

The other option is to keep the request paused - but I'm not sure if there are other scenarios which can lead to the request being freed.

Reply:
Updated to check if the req is null - this should be limited overhead.